src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1870,6 +1886,37 @@ protected override void ProcessRecord()
                         }
                     }
 
+                    // Handle the '-SecurityDescriptor' parameter
+                    RawSecurityDescriptor rawSecurityDescriptor = new RawSecurityDescriptor(SecurityDescriptor);
+                    RawAcl rawDiscretionaryAcl  = rawSecurityDescriptor.DiscretionaryAcl ;
+                    DiscretionaryAcl  discretionaryAcl   = new DiscretionaryAcl (false, false, rawDiscretionaryAcl );
+
+                    byte[] rawDacl = new byte[discretionaryAcl.BinaryLength];
+                    discretionaryAcl.GetBinaryForm(rawDacl, 0);
+                    rawSecurityDescriptor.DiscretionaryAcl = new RawAcl(rawDacl, 0);
+                    byte[] securityDescriptorByte = new byte[rawSecurityDescriptor.BinaryLength];
+                    rawSecurityDescriptor.GetBinaryForm(securityDescriptorByte, 0);
+
+                    if(!string.IsNullOrEmpty(SecurityDescriptor))
+                    {
+                        status = NativeMethods.SetServiceObjectSecurity(
+                                    hService,
+                                    SecurityInfos.DiscretionaryAcl,
+                                    securityDescriptorByte);
+                    }
+
+                    if (!status)
+                    {
+                        int lastError = Marshal.GetLastWin32Error();
+                        Win32Exception exception = new Win32Exception(lastError);
+                        WriteNonTerminatingError(
+                            service,
+                            exception,
+                            "CouldNotSetServiceSecurityDescriptor",
+                            ServiceResources.CouldNotSetServiceSecurityDescriptor,
+                            ErrorCategory.PermissionDenied);
```

Comment:
I am not sure about PermissionDenied - returned status code can be ERROR_INVALID_PARAMETER and others.

Reply:
How about InvalidOperation ?

Reply:
How about InvalidArgument ?

Reply:
I am thinking about some catch blocks not it is not request to change - only for discussion.